DiverseMarket.Backend/Services/ProductService.cs

```diff
@@ -57,9 +61,84 @@ public static List<ProductOfferBasicInfoDTO> GetAllProductOffersByCompanyUserId(
+                int productId = ProductDB.InsertProduct(product);
+                ProductOfferBasicInfoDTO productOfferDTO = new ProductOfferBasicInfoDTO(productOfferInsertDTO.CompanyUserId, productId, productOfferInsertDTO.Price, productOfferInsertDTO.Quantity);
+                return ProductOfferDB.InsertProductOffer(productOfferDTO);
+            }
+            else if (productOfferInsertDTO.Category.Equals(ProductCategory.Calcados.ToString()))
+            {
+                Model.Product product = new Model.Product(productOfferInsertDTO.Name, productOfferInsertDTO.Description, (long)ProductCategory.Calcados);
+                int productId = ProductDB.InsertProduct(product);
+                ProductOfferBasicInfoDTO productOfferDTO = new ProductOfferBasicInfoDTO(productOfferInsertDTO.CompanyUserId, productId, productOfferInsertDTO.Price, productOfferInsertDTO.Quantity);
+                return ProductOfferDB.InsertProductOffer(productOfferDTO);
+            }
+            else if (productOfferInsertDTO.Category.Equals(ProductCategory.Eletronicos.ToString()))
+            {
+                Model.Product product = new Model.Product(productOfferInsertDTO.Name, productOfferInsertDTO.Description, (long)ProductCategory.Eletronicos);
+                int productId = ProductDB.InsertProduct(product);
+                ProductOfferBasicInfoDTO productOfferDTO = new ProductOfferBasicInfoDTO(productOfferInsertDTO.CompanyUserId, productId, productOfferInsertDTO.Price, productOfferInsertDTO.Quantity);
+                return ProductOfferDB.InsertProductOffer(productOfferDTO);
+            }
+            else if (productOfferInsertDTO.Category.Equals(ProductCategory.Livros.ToString()))
+            {
+                Model.Product product = new Model.Product(productOfferInsertDTO.Name, productOfferInsertDTO.Description, (long)ProductCategory.Livros);
+                int productId = ProductDB.InsertProduct(product);
+                ProductOfferBasicInfoDTO productOfferDTO = new ProductOfferBasicInfoDTO(productOfferInsertDTO.CompanyUserId, productId, productOfferInsertDTO.Price, productOfferInsertDTO.Quantity);
+                return ProductOfferDB.InsertProductOffer(productOfferDTO);
+            }
+            else if (productOfferInsertDTO.Category.Equals(ProductCategory.Joias.ToString()))
+            {
+                Model.Product product = new Model.Product(productOfferInsertDTO.Name, productOfferInsertDTO.Description, (long)ProductCategory.Joias);
+                int productId = ProductDB.InsertProduct(product);
+                ProductOfferBasicInfoDTO productOfferDTO = new ProductOfferBasicInfoDTO(productOfferInsertDTO.CompanyUserId, productId, productOfferInsertDTO.Price, productOfferInsertDTO.Quantity);
+                return ProductOfferDB.InsertProductOffer(productOfferDTO);
+            }
+            else
+            {
+                return false;
+            }
+
+ 
+        }
```

Comment:
Dá pra simplificar aqui também: 

```suggestion
 public static bool InsertNewProductWithProductOffer(ProductOfferInsertDTO productOfferInsertDTO)
 {
     try
     {
         var product = new Model.Product(productOfferInsertDTO.Name, productOfferInsertDTO.Description, (long)Enum.Parse<ProductCategory>(productOfferInsertDTO.Category));
         int productId = ProductDB.InsertProduct(product);
         var productOfferDTO = new ProductOfferBasicInfoDTO(productOfferInsertDTO.CompanyUserId, productId, productOfferInsertDTO.Price, productOfferInsertDTO.Quantity);
         return ProductOfferDB.InsertProductOffer(productOfferDTO);
     } catch (Exception e)
     {
         new LogMessage(e);
         return false;
     }
 }
```